includes/class-wc-subscription.php

```diff
@@ -1316,7 +1327,7 @@ public function get_date_to_display( $date_type = 'next_payment' ) {
 				// translators: placeholder is human time diff (e.g. "3 weeks")
 				$date_to_display = sprintf( __( '%s ago', 'woocommerce-subscriptions' ), human_time_diff( current_time( 'timestamp', true ), $timestamp_gmt ) );
 			} else {
-				$date_to_display = date_i18n( wc_date_format(), $this->get_time( $date_type, 'site' ) );
+				$date_to_display = date_i18n( wc_date_format(), $timestamp_gmt + wc_timezone_offset() );
```

Comment:
This change removes one of the `get_time()` calls. Given we have the timestamp in GMT, and we want the time in the site's timezone, we offset the GMT time by the timezone offset instead.